ccmbioinfo/lfs
# CARDS-2372: Add support for data migrators

Add a data migrator that tracks the previous version and checks if any created data migrators need to run.

Also includes CARDS-2336.

To test:
- build and run an older version of cards:
- `git checkout cards-0.9.15`
- `mvn clean install -P docker`
- `cd compose-cluster`
- `python3 generate_compose_yaml.py --dev_docker_image  --composum --cards_project cards4heracles --mongo_singular`
- `docker-compose build && docker-compose up`
- Create a few CPET Interpretation forms, fill in the Protocol question with a few Bike value and a few other values
- Stop docker
- `cd ..`
- `git checkout CARDS-2372`
- `mvn clean install -P docker`
- `cd compose-cluster`
- `python3 generate_compose_yaml.py --dev_docker_image  --composum --cards_project cards4heracles --mongo_singular`
- `docker-compose build && docker-compose up`
- Check that the previous Bike answers are now Bike 50 rpm and the previous other answers are unchanged
- Check that `localhost:8080/libs/cards/conf/PrevVersion.json` now has `0.9.0.SNAPSHOT` stored
